packages/automerge-repo/src/network/NetworkAdapter.ts

```diff
@@ -1,33 +1,83 @@
 import EventEmitter from "eventemitter3"
-import { PeerId, ChannelId } from "../types.js"
+import { SessionId } from "../EphemeralData.js"
+import { ChannelId, DocumentId, PeerId } from "../types.js"
 
 export abstract class NetworkAdapter extends EventEmitter<NetworkAdapterEvents> {
   peerId?: PeerId // hmmm, maybe not
 
   abstract connect(url?: string): void
 
-  abstract sendMessage(
-    peerId: PeerId,
-    channelId: ChannelId,
-    message: Uint8Array,
-    broadcast: boolean
-  ): void
+  abstract send(message: Message): void
 
   abstract join(): void
 
   abstract leave(): void
 }
 
+// utilities
+
+export function isValidMessage(message: Message): boolean {
```

Comment:
we should probably move these bits into a separate file; we can potentially also use it as a type validator

Reply:
Good idea. Done.